src/Microsoft.Health.Abstractions/Errors/ErrorMessage.cs

```diff
@@ -0,0 +1,50 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Abstractions.Errors
+{
+    public class ErrorMessage
+    {
+        /// <summary>
+        /// A unique identifier for the ErrorMessage
+        /// <remarks>
+        /// This Id can be generated by the caller, or left null.
+        /// If null, the IErrorMessageService will generate an Id.
```

Comment:
nit:
```suggestion
        /// If null, the <see cref="IErrorMessageService" /> will generate an Id.
```

Reply:
Although, I am unsure you want to so tightly bound this class with the service my mentioning it explicitly here